src/Native/Unix/configure.cmake

```diff
@@ -174,6 +174,20 @@ check_cxx_source_compiles(
 
 check_cxx_source_compiles(
     "
+    #include <dirent.h>
+    int main(void)
+    {
+	DIR* dir;
+        struct dirent* entry;
+	struct dirent* result;
+        readdir_r(dir,entry,&result);
+	return 0;
+    }
+    "
+    HAVE_GNU_READDIR_R)
```

Comment:
I'm a bit surprised things compiled successfully in CI without your also adding an entry to the https://github.com/dotnet/corefx/blob/master/src/Native/Unix/Common/pal_config.h.in file.  If for no other reason than consistency, it'd be good to add this there, too.